*** edison-image.bb.orig	2016-03-24 11:15:44.818433703 +0100
--- edison-image.bb	2016-03-24 11:25:23.494444567 +0100
***************
*** 37,43 ****
  inherit bootimg
  do_bootimg[depends] += "${PN}:do_rootfs"
  
! IMAGE_ROOTFS_SIZE = "524288"
  
  IMAGE_FEATURES += "package-management ssh-server-openssh"
  # Allow passwordless root login and postinst logging
--- 37,43 ----
  inherit bootimg
  do_bootimg[depends] += "${PN}:do_rootfs"
  
! IMAGE_ROOTFS_SIZE = "1048576"
  
  IMAGE_FEATURES += "package-management ssh-server-openssh"
  # Allow passwordless root login and postinst logging
***************
*** 168,170 ****
--- 168,204 ----
  # nfs
  IMAGE_INSTALL += "nfs-utils"
  
+ IMAGE_INSTALL += "qtbase"
+ 
